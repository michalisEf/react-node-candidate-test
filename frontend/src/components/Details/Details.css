.details {
  & div {
    font-size: 1rem;
    line-height: 1.5;
  }
}
