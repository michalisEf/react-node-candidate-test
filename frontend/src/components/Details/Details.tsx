import React from 'react';

import { IDetails } from '../../interfaces';

import './Details.css';

const getSalaryString = (salary: number): string => {
	const formatter = new Intl.NumberFormat('en-GB', {
		style: 'currency',
		currency: 'GBP',
	});
	return formatter.format(salary).replace('.00', '');
};

const Details = ({ salary, type, location, company }: IDetails): JSX.Element => {
	const salaryString = getSalaryString(salary);
	return (
		<div className="details">
			<div>{location}</div>
			<div>{salaryString}</div>
			<div>{type}</div>
			<div>{company}</div>
		</div>
	);
};

export default Details;
