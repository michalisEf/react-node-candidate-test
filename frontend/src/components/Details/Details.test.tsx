import React from 'react';
import { render, screen } from '@testing-library/react'

import Details from './Details';

it('renders Details component', () => {
	const props = {
		location: 'London',
		salary: 50000,
		type: 'Permanent',
		company: 'Acme'
	};
	render(<Details { ...props }/>);
	const details = screen.getByText('London');
	expect(details).toBeInTheDocument();
	expect(details).toMatchSnapshot();
});
