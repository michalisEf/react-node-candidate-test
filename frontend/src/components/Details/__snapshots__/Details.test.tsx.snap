// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Details component 1`] = `
<div>
  London
</div>
`;
